### DISCUSSION TITLE
Simpler way to directly upload to Shared Albums

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#931] [Enhancements]
Only uploaded photos can be added to shared albums currently.

So if a user wants to adds a new photo to a shared album, we expect them to first upload that photo to one of their albums, and then add to the shared album.

Now with the introduction of `Uncategorized`, we can simplify this flow.

We should let users add photos to the shared albums directly, while in the background we first upload the items to their `Uncategorized` album, and then add a reference to it to the shared album.

#### Ref. 
https://github.com/ente-io/photos-app/blob/6b5db665aa56596ed1fc63b5941572741b752f76/lib/services/hidden_service.dart#L46

**Comment by TheGroundZero** [comment]
It appears like it is still not possible to (with a single action) upload pictures (or folders) to a shared album?
I'm attempting to migrate multiple albums from other cloud services and some of the target albums where created by my partner and shared with me. Now I cannot upload my back-ups to those albums.
The way I read the above, I need to upload them al to `Uncategorized`, then find them between the 10k+ Uncategorized pictures (no idea how it became that many), and them move them to the shared album?

**Comment by AswinAsok** [comment]
Closing this dicusssion, as files can now be uploaded directly to shared albums from all clients.

Mobile already supported this and with the latest Desktop release (v1.7.24), the same workflow is now available on Web and Desktop as well.

Users will just have to upload to their shared album and the the app in the background would first upload the items to their Uncategorized album, and then add a reference to it to the shared album.
